### PR TITLE
Use _PyLong_GetOne() and _PyLong_GetZero() in long_invmod()

### PR DESCRIPTION
**Pull request by vstinner**
These functions cannot fail.

### CODE REVIEW

**Review by corona10 — APPROVED** [review]
lgtm

**Review by kumaraditya303 — APPROVED** [review]

**Comment by vstinner** [comment]
Merged, thanks for reviews!

File: Objects/longobject.c [hunk 1]
```diff
@@ -4828,21 +4828,12 @@ long_divmod(PyObject *a, PyObject *b)
...
-    if (b == NULL) {
-        return NULL;
-    }
-    c = (PyLongObject *)PyLong_FromLong(0L);
-    if (c == NULL) {
-        Py_DECREF(b);
-        return NULL;
-    }
     Py_INCREF(a);
+    PyLongObject *b = (PyLongObject *)Py_NewRef(_PyLong_GetOne());
```
Comment (eendebakpt) [inline Objects/longobject.c:4835]: @vstinner Out of curiosity: why did you add the `Py_NewRef` when `_PyLong_GetOne` is documented to return an immortal singleton?
(or can modern compilers optimize this away?)

Comment (vstinner) [reply Objects/longobject.c:4835]: The Py_NewRef() is useless. I added it because the code wants a strong refererence and I didn't pay attention that these two constants are immortal.

>  (or can modern compilers optimize this away?)

No, Py_NewRef(<immortal>) is not optimized.

See also my PR gh-124076 attempt.
